### PR TITLE
Add nyc-taxi dataset download uilities

### PR DESCRIPTION
**Pull request by fsaintjacques**

### CODE REVIEW

**Comment by wesm** [comment]
This is an interesting choice for a location for this code, what are you thoughts about alternatives?

**Comment by fsaintjacques** [comment]
I could move it into arrow/examples/ would you prefer this?

**Comment by wesm** [comment]
If the purpose of this code is to generate example datasets for Arrow development, then this might make more sense. I could imagine having a data generation utility that could be maintained to create various testing datasets for use across implementations

**Comment by wesm** [comment]
Others might have other opinions though...

File: data/nyc-taxi/download2parquet.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    "passenger_count": "int8",
+    "trip_distance": "float32",
+    "pickup_longitude": "float32",
+    "pickup_latitude": "float32",
+    "pickup_location_id": "int32",
+    "dropoff_longitude": "float32",
+    "dropoff_latitude": "float32",
+    "dropoff_location_id": "int32",
+    "rate_code_id": "str",
+    "store_and_fwd_flag": "str",
```
Comment (xhochy) [inline data/nyc-taxi/download2parquet.py]: This should be boolean.

**Comment by xhochy** [comment]
I would also put this into `arrow/examples/`, I consider `parquet-testing` to be a binary file only repository.

**Comment by fsaintjacques** [comment]
Ok, I'll move it there!

**Pull request closed without merging**